src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs

```diff
@@ -255,14 +255,19 @@ public static IHostBuilder ConfigureDefaults(this IHostBuilder builder, string[]
                 });
 
             })
-            .UseDefaultServiceProvider((context, options) =>
+.UseDefaultServiceProvider((context, options) =>
             {
                 bool isDevelopment = context.HostingEnvironment.IsDevelopment();
                 options.ValidateScopes = isDevelopment;
                 options.ValidateOnBuild = isDevelopment;
             });
 
             return builder;
+
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+                Justification = "Calling IConfiguration.GetValue is safe when the T is bool.")]
```

Comment:
I wonder if it would be valuable to add some new overloads to this method for common things, like `bool`, `int`, and `string`. That way we don't need to suppress them in all the callsites.
It would obviously need to go through API review, but I think it might be worth it to add.

Reply:
I agree that it would be valuable for our trimming purposes, but not so sure if that is good enough value to add the extra overloads through API Review. Anyway, I'll log an issue to track this proposal and let area-owners/ApiProposal board to decide.

Reply:
We did for JsonValue: https://github.com/dotnet/runtime/issues/52773. The generic overload was unsafe, but we added overloads for the primitive values because they are "safe".